### ISSUE TITLE
Execute API calls from UI

### ISSUE THREAD

**Issue by nilkanth987** [asyncapi/asyncapi-react#914]
#### Reason/Context
It would make frontend devs life easier if they could execute the API directly from the generated documentation just like in Swagger. 

#### Description
We have to show an Execute button and on its click it should show the server and authorisation settings and call the API from the doc itself.
We also have to show the API response in the UI.

**Comment by derberg** [comment]
Moved the issue as html-template is basically a wrapper for the react component where all feature like this need to be added

**Comment by Papooch** [comment]
Is this actually something that _can_ be easily supported? The web client would have to act as a real publisher/subscriber and be able to connect to an actual message broker like Kafka or RabbitMQ. The only way to support this that I can think of is via an extra server component and websocket communication.

**Comment by AryanDesh** [comment]
I also thought of having a websocket connection but allowing your frontend to access to a backend server even just for adding tasks will become a vulnerability

**Comment by nightknighto** [comment]
I would love to indeed be able to interact with brokers and simulate sending or receiving messages through the API documentation. Kind of expected it to be already implemented similar to Swagger. This will be super helpful for testing, debugging, and development.

Is this issue accepted and waiting for a contribution, or is it still under validation?

**Comment by Shepard2154** [comment]
> Moved the issue as html-template is basically a wrapper for the react component where all feature like this need to be added
@derberg , what do you think of the following suggestions?:

- I can try to create a form and a "Try it out" button (as in Swagger) with parameters, payload and header transmission (for a start, perhaps in raw json format).
- then I can implement the usual POST method from the web form to the server (for example, at the same URL where the documentation is deployed). so that the server side can interact with brokers directly with data from the form.
- and maybe add the ability to configure sending data to a real broker or to a test broker (just to RAM)

I imagine something like this:

![Image](https://github.com/user-attachments/assets/56f88606-9fd0-49da-ad18-c1eb936b2990)
![Image](https://github.com/user-attachments/assets/12a29362-f766-4c54-a233-df4a5b7a4d91)

**Comment by Lancetnik** [comment]
@AceTheCreator, hello! Can you please help @Shepard2154 find the right direction?

**Comment by Lancetnik** [comment]
@AceTheCreator hi! The problem still actual, can you help us here, please?

**Comment by catosaurusrex2003** [comment]
Love the idea of a “Try it” in asyncapi-react.
One caveat from the browser side. We cannot connect to many AsyncAPI protocols from a web page because browsers only support HTTP, WebSocket, SSE, and WebRTC data channels. That means classic clients for MQTT over TCP, AMQP, Kafka, and similar are not available in the browser. The usual workaround is a proxy or gateway that exposes WS or HTTP and translates to the target protocol on the server network.

First class support is realistic only for servers that expose WebSocket or HTTP endpoints. Example: WSS and MQTT over WSS.
I am willing to work on the WebSocket part of this.

What is the plan for protocols that do not have a WS or HTTP surface by default, such as AMQP and Kafka?

**Comment by AceTheCreator** [comment]
Hey @Lancetnik 👋🏽

I've introduced a plugin system for the React component that enables developers to build and manage custom plugins independently.

Check out the [current implementation](https://github.com/asyncapi/asyncapi-react/pull/1185/files#diff-1365c9a6009af2b85ba6004a8dcc580c4e29774611e40d5c0497ec0d6b47babc) to get started. Feel free to test it out before the PR merges.

Once this PR is merged, we'll open a follow-up PR for dynamic plugin registration via refs(works with callback) and CDN-based plugin loading.

**Comment by Aditya-twd** [comment]
@AceTheCreator 
I'm interested in working on this feature! I've reviewed the plugin system you introduced in #1185 and I think it's a great approach for this feature. I'd love to contribute and help build out the HTTP/WebSocket support for the "try it out" form. Looking forward to hearing your thoughts!
